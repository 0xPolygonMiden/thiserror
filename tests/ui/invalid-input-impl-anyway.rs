use thiserror::Error;

#[derive(Error, Debug)]
#[error]
pub struct MyError;

fn main() {
    // FIXME: there should be no error on the following line. Thiserror should
    // emit an Error impl regardless of the bad attribute.
    _ = &MyError as &dyn std::error::Error;
}
