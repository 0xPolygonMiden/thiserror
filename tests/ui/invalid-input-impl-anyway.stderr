error: expected attribute arguments in parentheses: #[error(...)]
 --> tests/ui/invalid-input-impl-anyway.rs:4:3
  |
4 | #[error]
  |   ^^^^^

error[E0277]: the trait bound `MyError: std::error::Error` is not satisfied
  --> tests/ui/invalid-input-impl-anyway.rs:10:9
   |
10 |     _ = &MyError as &dyn std::error::Error;
   |         ^^^^^^^^ the trait `std::error::Error` is not implemented for `MyError`
   |
   = note: required for the cast from `&MyError` to `&dyn std::error::Error`
